pkg/kubelet/kubelet.go

```diff
@@ -2613,3 +2613,23 @@ func isSyncPodWorthy(event *pleg.PodLifecycleEvent) bool {
 	// ContainerRemoved doesn't affect pod state
 	return event.Type != pleg.ContainerRemoved
 }
+
+// PrepareDynamicResources calls container Manager PrepareDynamicResources API
+// This method implements RuntimeHelper interface
+func (kl *Kubelet) PrepareDynamicResources(pod *v1.Pod) error {
+	//if kl.containerManager != nil {
+	if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
+		return kl.containerManager.PrepareDynamicResources(pod)
+	}
+	return nil
+}
+
+// UnprepareDynamicResources calls container Manager UnprepareDynamicResources API
+// This method implements RuntimeHelper interface
+func (kl *Kubelet) UnprepareDynamicResources(pod *v1.Pod) error {
+	//if kl.containerManager != nil {
+	if utilfeature.DefaultFeatureGate.Enabled(features.DynamicResourceAllocation) {
```

Comment:
I think we should remove the check for the feature gate here and move it outside to any callers of this API.